dask/dataframe/tests/test_shuffle.py

```diff
@@ -916,3 +916,9 @@ def test_set_index_timestamp():
 
     assert_eq(df2, ddf_new_div)
     assert_eq(df2, ddf.set_index("A"))
+
+
+@pytest.mark.parametrize("compression", [None, "ZLib"])
+def test_disk_shuffle_with_compression(compression):
+    with dask.config.set({"shuffle-disk-compression": compression}):
+        test_shuffle("disk")
```

Comment:
How can we check that compression is actually being used?

Reply:
We could check the (hash) of the binary output of partd (or the file size) against a test value for a given dataframe. However I have no idea if partd or dask always returns fully deterministic output. I think this is not the case for partitions. Also dask generates a couple of uuid file tokens which change per run.
But I do not know the library deep enough for that.

Reply:
I think checking the hashed values makes sense. Do any compression formats include a known header or footer that we can check for the presence of?

Reply:
We probably can only use zlib or bz2 as snappy and blosc are not part of the standard python setup. Both have something like a 4 byte header but it depends on the version and the settings of the compression algorithm. Additionally, this would be like more or less a hacky test.

Can we somehow generate an deterministic output from both partd and dask?
OR: we can simply make a test and compress it with zlib and try if decompression fails. Furthermore: is there a test which checks the consistency of uncompressed partd files? [Side note: the dask project is quite sparingly when it comes to comments, especially in the tests]

Reply:
I'm not sure.

For this test, I'd recommend getting the file written to disk by dask and attempting to read it with the configured compression algorithm. That not raising should be sufficient to say that the config value was respected.

Reply:
I added an test which compares the raw disk size for compressed and uncompressed partd objects. Which at leasts checks if the compression flag was respected.

I was not able to:

- decompress the raw file with the compression algorithm as partd adds some kind of header or uses pickle?

- open the file with another partd object and re-read the data (due to file locking?)

In fact I gave up there as I could not figure out the internals of the partd library due lack of in-code documentation and time.

Reply:
I think the test you have later on (with a small modification) will be sufficient, thanks.